.github/workflows/govulncheck.yml

```diff
@@ -0,0 +1,150 @@
+
+env:
+  GO_VERSION: "~1.18"
+  BRIDGE_SERVER_ARTIFACT_PREFIX: "BRIDGE_SERVER"
+  BRIDGE_ARTIFACT_PREFIX: "BRIDGE"
+  API_ARTIFACT_PREFIX: "API"
+  APPROVAL_SVC_ARTIFACT_PREFIX: "APPROVAL_SVC"
+  DISTRIBUTOR_ARTIFACT_PREFIX: "DISTRIBUTOR"
+  SHIPYARD_CONTROLLER_ARTIFACT_PREFIX: "SHIPYARD_CONTROLLER"
+  SECRET_SVC_ARTIFACT_PREFIX: "SECRET_SVC"
+  RESOURCE_SVC_ARTIFACT_PREFIX: "RESOURCE_SVC"
+  REMEDIATION_SVC_ARTIFACT_PREFIX: "REMEDIATION_SVC"
+  LIGHTHOUSE_SVC_ARTIFACT_PREFIX: "LIGHTHOUSE_SVC"
+  MONGODB_DS_ARTIFACT_PREFIX: "MONGODB_DS"
+  STATISTICS_SVC_ARTIFACT_PREFIX: "STATISTICS_SVC"
+  WEBHOOK_SVC_ARTIFACT_PREFIX: "WEBHOOK_SVC"
+
+  BRIDGE_ARTIFACT: "bridge2"
+  BRIDGE_FOLDER: "bridge/"
+  BRIDGE_DOCKER_TEST_TARGET: "builder-test-unit"
+
+  BRIDGE_SERVER_ARTIFACT: "bridge2"
+  BRIDGE_SERVER_FOLDER: "bridge/"
+  BRIDGE_SERVER_DOCKER_TEST_TARGET: "bridge-server-test"
+  BRIDGE_SERVER_SHOULD_PUSH_IMAGE: "false"
+
+  API_ARTIFACT: "api"
+  API_FOLDER: "api/"
+  API_DOCKER_TEST_TARGET: "builder-test"
+
+  APPROVAL_SVC_ARTIFACT: "approval-service"
+  APPROVAL_SVC_FOLDER: "approval-service/"
+  APPROVAL_SVC_DOCKER_TEST_TARGET: "builder-test"
+
+  DISTRIBUTOR_ARTIFACT: "distributor"
+  DISTRIBUTOR_FOLDER: "distributor/"
+  DISTRIBUTOR_DOCKER_TEST_TARGET: "builder-test"
+
+  SHIPYARD_CONTROLLER_ARTIFACT: "shipyard-controller"
+  SHIPYARD_CONTROLLER_FOLDER: "shipyard-controller/"
+  SHIPYARD_CONTROLLER_DOCKER_TEST_TARGET: "builder-test"
+
+  SECRET_SVC_ARTIFACT: "secret-service"
+  SECRET_SVC_FOLDER: "secret-service/"
+  SECRET_SVC_DOCKER_TEST_TARGET: "builder-test"
+
+  RESOURCE_SVC_ARTIFACT: "resource-service"
+  RESOURCE_SVC_FOLDER: "resource-service/"
+  RESOURCE_SVC_DOCKER_TEST_TARGET: "builder-test"
+
+  REMEDIATION_SVC_ARTIFACT: "remediation-service"
+  REMEDIATION_SVC_FOLDER: "remediation-service/"
+  REMEDIATION_SVC_DOCKER_TEST_TARGET: "builder-test"
+
+  LIGHTHOUSE_SVC_ARTIFACT: "lighthouse-service"
+  LIGHTHOUSE_SVC_FOLDER: "lighthouse-service/"
+  LIGHTHOUSE_SVC_DOCKER_TEST_TARGET: "builder-test"
+
+  MONGODB_DS_ARTIFACT: "mongodb-datastore"
+  MONGODB_DS_FOLDER: "mongodb-datastore/"
+  MONGODB_DS_DOCKER_TEST_TARGET: "builder-test"
+
+  STATISTICS_SVC_ARTIFACT: "statistics-service"
+  STATISTICS_SVC_FOLDER: "statistics-service/"
+  STATISTICS_SVC_DOCKER_TEST_TARGET: "builder-test"
+
+  WEBHOOK_SVC_ARTIFACT: "webhook-service"
+  WEBHOOK_SVC_FOLDER: "webhook-service/"
+  WEBHOOK_SVC_DOCKER_TEST_TARGET: "builder-test"
+
+defaults:
+  run:
+    shell: bash
+jobs:
+  prepare:
+    name: Prepare govulncheck run
+    runs-on: ubuntu-22.04
+    outputs:
+      BUILD_MATRIX: ${{ steps.check_modified_files.outputs.BUILD_MATRIX }}
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
+
+    - name: Prepare artifact build matrix
+      id: check_modified_files
+      env:
+        BUILD_EVERYTHING: true
+      run: ./gh-actions-scripts/prepare_changed_artifacts.sh ""
```

Comment:
You could use a much simpler matrix that doesn't even need to be dynamically generated here...